crates/wasmparser/src/validator/operators.rs

```diff
@@ -4124,6 +4128,115 @@ where
         self.pop_operand(Some(ValType::Ref(RefType::I31REF)))?;
         self.push_operand(ValType::I32)
     }
+    fn visit_try(&mut self, mut ty: BlockType) -> Self::Output {
+        if !self.features.legacy_exceptions() {
```

Comment:
Oh this is automatically checked via the macro above so you can skip the check on `self.features` here and below

Reply:
Hm.. then these tests fail:

[test-01.log](https://github.com/user-attachments/files/16195427/test-01.log)

Reply:
Maybe it is because now, to parse legacy exceptions we need to set all or two flags:
```sh
❯ cargo build
❯ ./target/release/wasm-tools validate -fexception-handling,legacy-exceptions tests/local/legacy-exceptions.wat
```

Reply:
Ah for that you'll need to update [this exception list](https://github.com/bytecodealliance/wasm-tools/blob/021389c149dc7fa0c0bac97c0bd84ffe4986d889/tests/roundtrip.rs#L646-L663)

Reply:
You additionally might need to [remove legacy-exceptions from the default list](https://github.com/bytecodealliance/wasm-tools/blob/021389c149dc7fa0c0bac97c0bd84ffe4986d889/tests/roundtrip.rs#L578-L582) and then add a [special case](https://github.com/bytecodealliance/wasm-tools/blob/021389c149dc7fa0c0bac97c0bd84ffe4986d889/tests/roundtrip.rs#L627-L629) where it is enabled.

Reply:
Should I also update https://github.com/bytecodealliance/wasm-tools/blob/021389c149dc7fa0c0bac97c0bd84ffe4986d889/crates/wasmparser/src/lib.rs#L161-L165 to this:
```
// Deprecated old instructions from the exceptions proposal
@legacy_exceptions Try { blockty: $crate::BlockType } => visit_try
@legacy_exceptions Catch { tag_index: u32 } => visit_catch
@legacy_exceptions Rethrow { relative_depth: u32 } => visit_rethrow
@legacy_exceptions Delegate { relative_depth: u32 } => visit_delegate
@legacy_exceptions CatchAll => visit_catch_all
```
?

Reply:
Oh sorry yes definitely!

Reply:
After add `"legacy-exceptions.wat" => features.insert(WasmFeatures::LEGACY_EXCEPTIONS),` and other fixes, some tests still failed:
[test-02.log](https://github.com/user-attachments/files/16195973/test-02.log)

Reply:
To be clear, I pushed all changes now (tests not pass locally).

Reply:
That's [this comment](https://github.com/bytecodealliance/wasm-tools/pull/1661#discussion_r1676024130), notably updating [this list](https://github.com/bytecodealliance/wasm-tools/blob/021389c149dc7fa0c0bac97c0bd84ffe4986d889/tests/roundtrip.rs#L646-L663)